pset/blinder.go

```diff
@@ -113,6 +136,178 @@ func (b *blinder) BlindOutputs() error {
 	return nil
 }
 
+//BlindOutputs method blinds pset's issuance pseudo input
+func (b *blinder) blindInputs() error {
+	for index, input := range b.pset.UnsignedTx.Inputs {
+		if input.Issuance != nil {
+			if b.issuanceBlindingPrivateKeys == nil {
+				return nil
+			}
+
+			newIssuance := transaction.NewTxIssuanceFromContractHash(
+				input.Issuance.AssetEntropy,
+			)
+
+			err := newIssuance.GenerateEntropy(input.Hash, input.Index)
+			if err != nil {
+				return err
+			}
+
+			err = b.blindAsset(index, input, newIssuance)
+			if err != nil {
+				return err
+			}
+
+			if len(input.Issuance.TokenAmount) > 0 {
+				err = b.blindToken(index, input, newIssuance)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+	}
+	return nil
+}
+
+func (b *blinder) blindAsset(
+	index int,
+	input *transaction.TxInput,
+	newIssuance *transaction.TxIssuanceExtended,
+) error {
+	asset, err := newIssuance.GenerateAsset()
+	if err != nil {
+		return err
+	}
+
+	abf, err := generateRandomNumber()
+	if err != nil {
+		return err
+	}
+
+	assetAmount := input.Issuance.AssetAmount
+	assetCommitment, err := confidential.AssetCommitment(
+		asset,
+		abf,
+	)
+	if err != nil {
+		return err
+	}
+
+	vbf, err := generateRandomNumber()
+	if err != nil {
+		return err
+	}
+	amount := [9]byte{}
+	copy(amount[:], assetAmount)
+	assetAmountSatoshi, err := confidential.ElementsToSatoshiValue(amount)
+	valueCommitment, err := confidential.ValueCommitment(
+		assetAmountSatoshi,
+		assetCommitment[:],
+		vbf,
+	)
+	if err != nil {
+		return err
+	}
+
+	var vbf32 [32]byte
+	copy(vbf32[:], vbf)
+
+	var nonce [32]byte
+	copy(nonce[:], b.issuanceBlindingPrivateKeys[index].AssetKey[:])
+
+	rangeProofInput := confidential.RangeProofInput{
+		Value:               assetAmountSatoshi,
+		Nonce:               nonce,
+		Asset:               asset,
+		AssetBlindingFactor: abf,
+		ValueBlindFactor:    vbf32,
+		ValueCommit:         valueCommitment[:],
+		ScriptPubkey:        []byte{},
+		MinValue:            1,
+		Exp:                 0,
+		MinBits:             52,
+	}
+	rangeProof, err := confidential.RangeProof(rangeProofInput)
+	if err != nil {
+		return err
+	}
+
+	b.pset.UnsignedTx.Inputs[index].
+		Issuance.AssetAmount = assetCommitment[:]
```

Comment:
this should be set to the `valueCommitment`:
```suggestion
		Issuance.AssetAmount = valueCommitment[:]
```

Reply:
Done